File: gfapy/line/fragment/references.py
import gfapy

class References():

  def _initialize_references(self):
    s = self._gfa.segment(self.get("sid"))
    if s is None:
      if self._gfa._segments_first_order:
        raise gfapy.NotFoundError()
      s = gfapy.line.segment.GFA2({"sid": self.get("sid"),
                                   "slen": 1,
                                   "sequence": "*"},
                                   version = "gfa2",
                                   virtual = True)
      s.connect(self._gfa)
    self._set_existing_field("sid", s, set_reference = True)
    s._add_reference(self, "fragments")

File: lib/rgfa/line/fragment/references.rb
# Sets the reference to the segment in fragments, when they are connected
# to a RGFA instance; creates a virtual segment, if the segment has not
# been found yet
module RGFA::Line::Fragment::References

  private

  def initialize_references
    s = @rgfa.segment(get(:sid))
    if s.nil?
      raise RGFA::NotFoundError if @rgfa.segments_first_order
      s = RGFA::Line::Segment::GFA2.new({:sid => get(:sid),
                                         :slen => 1,
                                         :sequence => "*"},
                                         version: :gfa2,
                                         virtual: true)
      s.connect(@rgfa)
    end
    set_existing_field(:sid, s, set_reference: true)
    s.add_reference(self, :fragments)
  end

end
